android/libraries/rib-badoo/src/main/java/com/badoo/common/rib/BaseInteractor.kt

```diff
@@ -0,0 +1,89 @@
+package com.badoo.common.rib
+
+import android.arch.lifecycle.Lifecycle
+import android.arch.lifecycle.LifecycleOwner
+import android.arch.lifecycle.LifecycleRegistry
+import android.os.Bundle
+import android.support.annotation.CallSuper
+import com.badoo.mvicore.binder.Binder
+import com.badoo.mvicore.android.lifecycle.CreateDestroyBinderLifecycle
+import com.badoo.mvicore.android.lifecycle.StartStopBinderLifecycle
+import com.badoo.mvicore.android.lifecycle.ResumePauseBinderLifecycle
+import com.uber.rib.core.Interactor
+import com.uber.rib.core.RibView
+import io.reactivex.disposables.Disposable
+
+@Suppress("FINITE_BOUNDS_VIOLATION_IN_JAVA")
+abstract class BaseInteractor<V : RibView, R : BaseViewRouter<V, *>>(
+    private val disposables: List<Disposable>
+) : Interactor<R>(), LifecycleOwner {
+
+    private val ribLifecycleRegistry = LifecycleRegistry(this)
+    private val viewLifecycleRegistry = LifecycleRegistry(this)
+
+    final override fun didBecomeActive(savedInstanceState: Bundle?) {
+        super.didBecomeActive(savedInstanceState)
+        ribLifecycleRegistry.handleLifecycleEvent(Lifecycle.Event.ON_CREATE)
+        didBecomeActive(ribLifecycleRegistry, savedInstanceState)
+    }
+
+    open fun didBecomeActive(ribLifecycle: Lifecycle, savedInstanceState: Bundle?) {
+    }
+
+    @CallSuper
+    fun onViewCreated() {
```

Comment:
how about leaving `on` prefix just for internal callbacks? I would call it as `viewCreated`.
other possibility is to stick the existing RIBs rules and use `did` for the callbacks.